docs/notebooks/image_to_image_arcface.md

```diff
@@ -0,0 +1,224 @@
+0.27113
+INFO     Metric: 'resnet50_f1_score_at_k' before fine-tuning:  0.07631 after fine-tuning:     __main__.py:210
+0.35788
+INFO     Metric: 'resnet50_hit_at_k' before fine-tuning:  0.82900 after fine-tuning: 0.94100  __main__.py:210
+INFO     Metric: 'resnet50_average_precision' before fine-tuning:  0.52305 after fine-tuning: __main__.py:210
+0.79779
+INFO     Metric: 'resnet50_reciprocal_rank' before fine-tuning:  0.64909 after fine-tuning:   __main__.py:210
+0.89224
+INFO     Metric: 'resnet50_dcg_at_k' before fine-tuning:  1.30710 after fine-tuning: 4.52143  __main__.py:210
+INFO     Building the artifact ...                                                            __main__.py:215
+INFO     Pushing artifact to Jina AI Cloud ...                                                __main__.py:241
+[12:19:53] INFO     Artifact pushed under ID '63f8a9089c6406e19244771d'                                  __main__.py:243
+DEBUG    Artifact size is 83.580 MB                                                           __main__.py:245
+INFO     Finished 🚀                                                                          __main__.py:246
+```
+<!-- #endregion -->
+
+<!-- #region id="0l4e4GrspilM" -->
+## Saving
+
+After the run has finished successfully, you can download the tuned model on your local machine:
+
+<!-- #endregion -->
+
+```python id="KzfxhqeCmCa8"
+artifact = run.save_artifact('resnet-model')
+```
+
+<!-- #region id="gkNHTyBkprQ0" -->
+## Inference
+
+Now you saved the `artifact` into your host machine,
+let's use the fine-tuned model to encode a new `Document`:
+
+```{admonition} Inference with ONNX
+In case you set `to_onnx=True` when calling `finetuner.fit` function,
+please use `model = finetuner.get_model(artifact, is_onnx=True)`
+```
+<!-- #endregion -->
+
+```python id="bOi5qcNLplaI"
+query = DocumentArray([query_data[0]])
+
+model = finetuner.get_model(artifact=artifact, device='cuda')
+
+finetuner.encode(model=model, data=query)
+finetuner.encode(model=model, data=index_data)
+
+assert query.embeddings.shape == (1, 2048)
+```
+
+<!-- #region id="1cC46TQ9pw-H" -->
+And finally, you can use the embedded `query` to find top-k visually related images within `index_data` as follows:
+<!-- #endregion -->
+
+```python id="aYMnyr6ac4ln"
+query.match(index_data, limit=10, metric='cosine')
+```
+
+<!-- #region id="irvn0igWdLOf" -->
+## Before and after
+We can directly compare the results of our fine-tuned model with its zero-shot counterpart to get a better idea of how finetuning affects the results of a search. Each class of the Stanford cars dataset contains images for a single model of car. Therefore, we would expect the search results after finetuning to return more images of cars that are of the same model as the car in the query image. The improvement in results can be quite subtle for some queries, though the example below clearly shows that fine-tuning increases the quality of the search results
```

Comment:
this is a bit generic, maybe describe a little bit in which sense they differ etc. and what "quality" means here, e.g., retrieving cars of the same brand.